framework/web/UrlManager.php

```diff
@@ -431,22 +431,22 @@ public function createUrl($params)
             }
 
             if ($url !== false) {
-                if (strpos($url, '://') !== false) {
-                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 8)) !== false) {
+                if (Url::isRelative($url)) {
+                    $url = ltrim($url, '/');
+                    return "$baseUrl/{$url}{$anchor}";
+                } elseif (Url::isProtocolAgnostic($url)) {
+                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 2)) !== false) {
                         return substr($url, 0, $pos) . $baseUrl . substr($url, $pos) . $anchor;
                     }
 
                     return $url . $baseUrl . $anchor;
-                } elseif (strncmp($url, '//', 2) === 0) {
-                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 2)) !== false) {
+                } else {
+                    if ($baseUrl !== '' && ($pos = strpos($url, '/', 8)) !== false) {
```

Comment:
What is `8`?

Reply:
Looks like a typo, will check when back in the office.

Reply:
There was `8` in old implementation, but it is not clear why. There is also magic `2` above, which was relatively easy to understand before (previous condition for this `if` block was `strncmp($url, '//', 2) === 0`, so you can guess that your looking for part after `//`), but right now it is quite cryptic.